### ISSUE TITLE
Feature request: sync on edit

### ISSUE THREAD

**Issue by maxrothman** [zokugun/vscode-sync-settings#15]
This extension is great! Thank you for supporting a plain ol' directory as a sync target, unlike the VSCode settings sync feature.

It would be nice if I didn't have to trigger settings sync manually. Whenever I edit a setting or keybinding in the UI, or save the keybindings.json or settings.json files, the "Sync Settings: Upload (user -> repository)" command would be triggered. If it'd be undesirable for this to be enabled by default, it could be gated on a setting.

**Comment by daiyam** [comment]
If I implement it, it will be optional.

**Comment by Tealk** [comment]
I would also like to have this or a similar function. Automatic upload on time.

**Comment by daiyam** [comment]
@Tealk Can you open a new issue since yours is different from the original one 😉

**Comment by Tealk** [comment]
I can do that, but I don't think it's that different.

**Comment by daiyam** [comment]
It is because on edit, the extension would need to monitor for any changes done. While on time, it need to just sync.

**Comment by daiyam** [comment]
@Tealk You could use the newest feature:
```
"syncSettings.crons": {
    "review": "0 * * * *",      // every hour
}
```
It will check every hour if there is anything changes and prompt if you want to upload them.
